### PR TITLE
[PATCH v7] api: thread: clarify odp_thread_id() specification

### PR DESCRIPTION
**Pull request by MatiasElo**

### CODE REVIEW

File: test/common/odp_cunit_common.h [hunk 1]
```diff
@@ -80,7 +80,8 @@ int odp_cunit_run(void);
  * When 'priv' is 0, the same argument pointer (arg[0]) is passed to all threads. Otherwise,
  * a pointer is passed (from arg[]) to each thread. Returns 0 on success.
  */
-int odp_cunit_thread_create(int num, int func_ptr(void *arg), void *const arg[], int priv);
+int odp_cunit_thread_create(int num, int func_ptr(void *arg), void *const arg[],
+			    int priv, int sync);
```
Comment (TuomasTaipale) [inline test/common/odp_cunit_common.h:84]: Something could be said about what `sync` controls.

Comment (MatiasElo) [reply test/common/odp_cunit_common.h:84]: Sure, added comment in V2.

**Comment by MatiasElo** [comment]
V2:
- Documented `odp_cunit_thread_create()` function's new `sync` argument

**Review by TuomasTaipale — APPROVED** [review]

**Review by psavol — APPROVED** [review]

**Review by jerinjacobk — APPROVED** [review]

**Comment by MatiasElo** [comment]
V6:
- Rebase and reviewed-by tags added

**Pull request merged**